library/alloc/src/vec/mod.rs

```diff
@@ -1783,6 +1783,20 @@ impl<T, A: Allocator> Vec<T, A> {
     /// static_ref[0] += 1;
     /// assert_eq!(static_ref, &[2, 2, 3]);
     /// ```
+    ///
+    /// Drop excess capacity:
+    ///
+    /// ```
+    /// let mut v = Vec::with_capacity(10);
+    /// v.extend([1, 2, 3].iter().cloned());
+    /// let slice = v.clone().into_boxed_slice();
+    ///
+    /// unsafe {
+    ///     let p = slice.as_ptr();
+    ///     let rebuilt = std::slice::from_raw_parts(p, slice.len());
+    ///     assert_eq!(rebuilt.len(), 3);
+    /// }
```

Comment:
This example doesn't demonstrate any leaking. `.into_boxed_slice()` returns a `Box`, which the `slice` variable will own.

The issue this PR was originally trying to fix is that the documentation doesn't promise this function drops excess capacity. However, I don't think that's something we want to fix. Instead, I think it'd be better if leak explicitly does *not* drop the excess capacity. But for that, there's an RFC which has been standing still for a while: https://github.com/rust-lang/rfcs/pull/2969

Reply:
@m-ou-se Thank you for your support and help. I've been stucking and would like to close my PR.